src/apps/api/views/leaderboards.py

```diff
@@ -43,15 +44,24 @@ def update(self, request, *args, **kwargs):
         return response
 
 
+def is_user_submission_owner_or_admin(request, submission, competition):
```

Comment:
Let's put this logic in a permission class and stick it in `apps/api/permissions.py`. Then we can put it in the `permission_classes` tuple on our new, consolidated view.